### PR TITLE
Feature/dalamud window system

### PR DESCRIPTION
**Pull request by SheepGoMeh**
Migrate interface to the Dalamud windowing system.

* Moved service out of the Plugin class into their own class.
* Introduce static instance for Plugin class.
* Removed current interface implementation.
* Introduced a new interface implementation based on Dalamud.WindowSystem

### CODE REVIEW

File: HuntBuddy/Windows/ConfigurationWindow.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+{
+    public ConfigurationWindow() : base(
+        $"{Plugin.Instance.Name} configuration",
+        ImGuiWindowFlags.NoDocking,
+        true)
+    {
+        this.Size = Vector2.Zero;
+        this.SizeCondition = ImGuiCond.Always;
+        
+        if (Plugin.Instance.Configuration.LockWindowPositions)
```
Comment (PrincessRTFM) [inline HuntBuddy/Windows/ConfigurationWindow.cs:20]: This needs to go in the pre-draw function, or else it'll only ever set the flags when the plugin first loads and changing the settings won't have any effect live.

File: HuntBuddy/Windows/LocalHuntsWindow.cs [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+{
+    public LocalHuntsWindow() : base(
+        $"{Plugin.Instance.Name}",
+        ImGuiWindowFlags.NoNavInputs | ImGuiWindowFlags.NoDocking,
+        true)
+    {
+        this.Size = Vector2.Zero;
+        this.SizeCondition = ImGuiCond.Always;
+
+        if (Plugin.Instance.Configuration.HideLocalHuntBackground)
```
Comment (PrincessRTFM) [inline HuntBuddy/Windows/LocalHuntsWindow.cs:23]: This also needs to go in the pre-draw function, same issue as the config window

File: HuntBuddy/Windows/LocalHuntsWindow.cs [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        {
+            this.Flags |= ImGuiWindowFlags.NoBackground;
+        }
+
+        if (Plugin.Instance.Configuration.LockWindowPositions)
```
Comment (PrincessRTFM) [inline HuntBuddy/Windows/LocalHuntsWindow.cs:28]: Likewise here

File: HuntBuddy/Windows/MainWindow.cs [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+{
+    public MainWindow() : base(
+        $"{Plugin.Instance.Name}",
+        ImGuiWindowFlags.NoDocking,
+        true)
+    {
+        this.Size = new Vector2(400 * ImGui.GetIO().FontGlobalScale, 500);
+        this.SizeCondition = ImGuiCond.Once;
+
+        if (Plugin.Instance.Configuration.LockWindowPositions)
```
Comment (PrincessRTFM) [inline HuntBuddy/Windows/MainWindow.cs:24]: This also needs to go in the pre-draw function, same issue as the config window

**Review by PrincessRTFM — CHANGES_REQUESTED** [review]
Window classes need a few bugfixes; only two of the three are held in an instance variable in the main plugin class, not sure why. Otherwise, looks good.

File: HuntBuddy/Plugin.cs [hunk 5]
```diff
@@ -53,30 +33,47 @@ public class Plugin : IDalamudPlugin
...
+					(MobHuntStruct*)Service.SigScanner.GetStaticAddressFromSig(
 						"48 8D 0D ?? ?? ?? ?? 8B D8 0F B6 52");
 			}
+			
+			this.MainWindow = new MainWindow();
+			this.ConfigurationWindow = new ConfigurationWindow();
+			
+			this.WindowSystem = new WindowSystem("HuntBuddy");
+			this.WindowSystem.AddWindow(this.MainWindow);
+			this.WindowSystem.AddWindow(new LocalHuntsWindow());
```
[the remark this answers is not in the document]
Comment (PrincessRTFM) [reply HuntBuddy/Plugin.cs:69]: From discord discussion:
> I didn't see a use for it
> It's always set to open, and won't be rendered until all conditions are met

Fair enough.

File: HuntBuddy/Windows/ConfigurationWindow.cs (hunk 1 above)
Comment (SheepGoMeh) [reply HuntBuddy/Windows/ConfigurationWindow.cs:23]: Addressed in 93872f1

File: HuntBuddy/Windows/LocalHuntsWindow.cs (hunk 2 above)
Comment (SheepGoMeh) [reply HuntBuddy/Windows/LocalHuntsWindow.cs:28]: Addressed in 93872f1

File: HuntBuddy/Windows/LocalHuntsWindow.cs (hunk 3 above)
Comment (SheepGoMeh) [reply HuntBuddy/Windows/LocalHuntsWindow.cs:33]: Addressed in 93872f1

File: HuntBuddy/Windows/MainWindow.cs (hunk 4 above)
Comment (SheepGoMeh) [reply HuntBuddy/Windows/MainWindow.cs:27]: Addressed in 93872f1

**Review by PrincessRTFM — APPROVED** [review]
All issues addressed, looks good

**Pull request merged**